aqua-organisation/ah-django-Aqua
# #164691220 Add User profiles Functionality

#### What does this PR do?
Enables user to create a profile and also update it.

#### Description of Task to be completed?
*To get the following end points working*:
- ```**PUT** http://127.0.0.1:8000/api/user/```
- ```**GET** http://127.0.0.1:8000/api/user/``` 
- ```**GET** http://127.0.0.1:8000/api/profiles/<str:username>/```
- ```**GET** http://127.0.0.1:8000/api/profiles/```


#### How should this be manually tested?
**One** should be able to clone this repo and test out the code In Postman or any other tool of their choice to test it out.

- Added the relational model which creates a profile on the creation of a user.
- The user can update their profile if they have the right authorization using the route ```http://127.0.0.1:8000/api/user/ ``` 
The updated body should look like.
```
{
  "user":{
    "username": "nicksbros",
    "email": "some@some.com",
    "bio":"I work at a statefarm i",
    "image": "https://inspector"
    
}
}
```
- User can access their own profile using a get method using routes ```http://127.0.0.1:8000/api/user/``` or ``` http://127.0.0.1:8000/api/profiles/<str:username> ```
response should look like 
```
{
    "user": {
        "email": "some@some.com",
        "username": "nicksbros",
        "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6MSwidXNuIjoibmlja3Nicm9zIiwiZXhwIjoxNTU2OTk2MTkyfQ.W4XnFcVppUCfj6_vhF6TuGjXxoYCzbNDe1EeN6mP4E0",
        "bio": "I work at a statefarm i",
        "image": ""
    }
}
```
the image field shoul be black since it only returns an image object.

- The user should be able to view other people's profiles using route ``` http://127.0.0.1:8000/api/profiles/<username> ``` and the response object should look like 

``` {
    "profile": {
        "username": "nicksbrosty",
        "bio": "I am cool bro",
        "image": "https://static.productionready.io/images/smiley-cyrus.jpg"
    }
}
```
with a default route which represents the avatar to act as a place holder for the user's profile picture. 
- The user shouldn't be able to edit one's profile as for the PUT method for this route is blocked. like
```{
    "profile": {
        "detail": "Method \"PUT\" not allowed."
    }
}
```
- If user doesnt exist, response should look like ``` {
    "errors": {
        "detail": "The requested profile does not exist."
    }
}```


- A time stamp for created_at and updated_at


**Story Id** https://www.pivotaltracker.com/story/show/164691220
